cpp/src/Slice/Grammar.y

```diff
@@ -1765,112 +1782,81 @@ enumerator_initializer
-        {
-            pd->appendMetadata(std::move(metadata->v));
-        }
+        param = op->createParameter(tsp->name, tsp->type, tsp->isOptional, tsp->tag);
+        currentUnit->currentContainer()->checkIntroduced(tsp->name, param);
     }
+    $$ = param;
 }
-| out_qualifier metadata type keyword
+| type keyword
 {
-    auto isOutParam = dynamic_pointer_cast<BoolTok>($1);
-    auto type = dynamic_pointer_cast<Type>($3);
-    auto ident = dynamic_pointer_cast<StringTok>($4);
+    auto type = dynamic_pointer_cast<Type>($1);
+    auto ident = dynamic_pointer_cast<StringTok>($2);
+    ParameterPtr param;
+
     auto op = dynamic_pointer_cast<Operation>(currentUnit->currentContainer());
     if (op)
     {
-        op->createParameter(ident->v, type, isOutParam->v, false, 0); // Dummy
+        param = op->createParameter(ident->v, type, false, 0); // Dummy
         currentUnit->error("keyword '" + ident->v + "' cannot be used as parameter name");
     }
+    $$ = param;
 }
-| parameter_list ',' out_qualifier metadata type keyword
+| type
 {
-    auto isOutParam = dynamic_pointer_cast<BoolTok>($3);
-    auto type = dynamic_pointer_cast<Type>($5);
-    auto ident = dynamic_pointer_cast<StringTok>($6);
+    auto type = dynamic_pointer_cast<Type>($1);
+    ParameterPtr param;
+
     auto op = dynamic_pointer_cast<Operation>(currentUnit->currentContainer());
     if (op)
     {
-        op->createParameter(ident->v, type, isOutParam->v, false, 0); // Dummy
-        currentUnit->error("keyword '" + ident->v + "' cannot be used as parameter name");
+        param = op->createParameter(Ice::generateUUID(), type, false, 0); // Dummy
+        currentUnit->error("missing parameter name");
     }
+    $$ = param;
 }
-| out_qualifier metadata type
+| ICE_OUT parameter
 {
-    auto isOutParam = dynamic_pointer_cast<BoolTok>($1);
-    auto type = dynamic_pointer_cast<Type>($3);
-    auto op = dynamic_pointer_cast<Operation>(currentUnit->currentContainer());
-    if (op)
+    if (auto param = dynamic_pointer_cast<Parameter>($2))
     {
-        op->createParameter(Ice::generateUUID(), type, isOutParam->v, false, 0); // Dummy
-        currentUnit->error("missing parameter name");
+        param->setIsOutParam();
     }
+    $$ = $2;
 }
-| parameter_list ',' out_qualifier metadata type
+| local_metadata parameter
 {
-    auto isOutParam = dynamic_pointer_cast<BoolTok>($3);
-    auto type = dynamic_pointer_cast<Type>($5);
-    auto op = dynamic_pointer_cast<Operation>(currentUnit->currentContainer());
-    if (op)
+    if (auto param = dynamic_pointer_cast<Parameter>($2))
     {
-        op->createParameter(Ice::generateUUID(), type, isOutParam->v, false, 0); // Dummy
-        currentUnit->error("missing parameter name");
+        auto metadata = dynamic_pointer_cast<MetadataListTok>($1);
+        param->appendMetadata(std::move(metadata->v));
     }
+    $$ = $2;
 }
 ;
 
+// ----------------------------------------------------------------------
```

Comment:
This wasn't changed. It was just moved down.
Unfortunately git doesn't realize this.
But if you look under where `out_qualifier` used to be, you'll see it.